src/libsyncengine/syncpal/operationprocessor.cpp

```diff
@@ -91,90 +84,70 @@ std::shared_ptr<Node> OperationProcessor::correspondingNodeInOtherTree(std::shar
         }
     }
 
-    if (dbNodeId) {
-        // The node is in DB => find corresponding node id
-        NodeId nodeId;
-        bool found = false;
-        if (!_syncPal->_syncDb->id(otherSnapshotSide, *dbNodeId, nodeId, found)) {
-            LOG_SYNCPAL_WARN(_logger, "Error in SyncDb::id for dbNodeId=" << *dbNodeId);
-            return nullptr;
-        }
-        if (!found) {
-            LOG_SYNCPAL_WARN(_logger, "Node not found for dbNodeId = " << *dbNodeId);
-            return nullptr;
-        }
+    if (dbNodeId) return correspondingNodeDirect(node);
 
-        NodeId effectiveCorrespondingId = nodeId;
+    // The node is not in DB => find an ancestor
+    return findFromCommonAncestor(node);
+}
 
-        auto previousIdIt = otherTree->previousIdSet().find(nodeId);
-        if (previousIdIt != otherTree->previousIdSet().end()) {
-            effectiveCorrespondingId = previousIdIt->second;
-        }
+std::shared_ptr<Node> OperationProcessor::findFromCommonAncestor(std::shared_ptr<Node> node) {
+    ReplicaSide otherSnapshotSide =
+        (node->side() == ReplicaSide::ReplicaSideLocal ? ReplicaSide::ReplicaSideRemote : ReplicaSide::ReplicaSideLocal);
```

Comment:
I don't know how common is this pattern in the code. If it is very  common, a small helper function that switches replica sides would be cool to have.

Reply:
Just twice in this class. Do you think a helper function should be in this class or somewhere else? I never know where to implement those helpers

Reply:
It's really up to you. I would prefer to have one, so that the next time we need it, it will be much less typing. But there is no actual need at the moment.

Reply:
It's done 👍